File: zy-frame/src/main/java/com/sanjie/zy/http/reactive/TransformUtils.java
package com.sanjie.zy.http.reactive;


import com.sanjie.zy.rx.android.schedulers.AndroidSchedulers;

import io.reactivex.Observable;
import io.reactivex.ObservableSource;
import io.reactivex.ObservableTransformer;
import io.reactivex.schedulers.Schedulers;

/**
 * Created by LangSanJie on 2017/2/16.
 */

public class TransformUtils {

    public static <T> ObservableTransformer<T, T> defaultSchedulers() {
        return new ObservableTransformer<T, T>() {
            @Override
            public ObservableSource<T> apply(Observable<T> upstream) {
                return upstream.subscribeOn(Schedulers.io())
                        .observeOn(AndroidSchedulers.mainThread());
            }
        };
    }

    public static <T> ObservableTransformer<T, T> defaultNewThreadSchedulers() {
        return new ObservableTransformer<T, T>() {
            @Override
            public ObservableSource<T> apply(Observable<T> upstream) {
                return upstream.subscribeOn(Schedulers.newThread())
                        .observeOn(AndroidSchedulers.mainThread());
            }
        };
    }
}

File: zy-frame/src/main/java/com/sanjie/zy/rx/utils/TransformUtils.kt
package com.sanjie.zy.rx.utils

import com.sanjie.zy.rx.android.schedulers.AndroidSchedulers
import io.reactivex.ObservableTransformer
import io.reactivex.schedulers.Schedulers

/**
 * </p>
 * 创建者:SanJie
 * 时间:2019/8/6
 */
object TransformUtils {

    fun <T> defaultSchedulers(): ObservableTransformer<T, T> {
        return ObservableTransformer { upstream ->
            upstream.subscribeOn(Schedulers.io())
                .observeOn(AndroidSchedulers.mainThread())
        }
    }

    fun <T> defaultNewThreadSchedulers(): ObservableTransformer<T, T> {
        return ObservableTransformer { upstream ->
            upstream.subscribeOn(Schedulers.newThread())
                .observeOn(AndroidSchedulers.mainThread())
        }
    }
}